### PR TITLE
[NCL-7225] Implement Product Release create / edit page

### PR DESCRIPTION
**Pull request by patrikk0123**

### CODE REVIEW

**Comment by matedo1** [comment]
Hi @DnsZhou , do you think you could please review this one today?

**Comment by matedo1** [comment]
Hi @DnsZhou , please do you think you could review this one?

File: src/common/productReleaseEntityAttributes.ts [hunk 1]
```diff
@@ -20,6 +20,7 @@ export const productReleaseEntityAttributes = {
     filter: {
       operator: '=like=',
     },
+    tooltip: "Release part of the Product Version, e.g. '0.GA' for Product ersion '1.0'.",
```
Comment (DnsZhou) [inline src/common/productReleaseEntityAttributes.ts:23]: typo
`tooltip: "Release part of the Product Version, e.g. '0.GA' for Product version '1.0'.",`

File: src/common/productReleaseEntityAttributes.ts [hunk 2]
```diff
@@ -38,4 +39,15 @@ export const productReleaseEntityAttributes = {
...
+  commonPlatformEnumeration: {
+    id: 'commonPlatformEnumeration',
+    title: 'Common Platform Enumeration (CPE)',
+    tooltip:
+      "Common Platform Enumeration associated with the Product Release. CPEs are used to map packages that are security-relevant and delivered via security errata back to products and by the CVE Engine to map errata to containers when grading (e.g. 'cpe:/a:redhat:jboss_data_grid:7.3.5'). Can be retrieved from the Product Pages.",
+  },
+  productPagesCode: {
+    id: 'productPagesCode',
+    title: 'Product Pages Code',
+    tooltip: "Code associated with the Product Release in the Product Pages (e.g. 'rhdg-7-3.5').",
```
Comment (DnsZhou) [inline src/common/productReleaseEntityAttributes.ts:51]: Mind the preposition.
`tooltip: "Code associated with the Product Release on the Product Pages (e.g. 'rhdg-7-3.5').",`

File: src/components/ProductVersionReleasesList/ProductVersionReleasesList.tsx
Comment (DnsZhou) [inline src/components/ProductVersionReleasesList/ProductVersionReleasesList.tsx:1]: Consider the sorting feature especially sorting by release date will be helpful.

File: src/components/ProductReleaseCreateEditPage/ProductReleaseCreateEditPage.tsx [hunk 3]
```diff
@@ -0,0 +1,368 @@
...
+          throw new PncError({
+            code: 'NEW_ENTITY_ID_ERROR',
+            message: `Invalid productReleaseId coming from Orch POST response: ${newProductReleaseId}`,
+          });
+        }
+
+        navigate('..');
+      })
+      .catch((error: any) => {
+        console.error('Failed to create Product Release.');
```
Comment (DnsZhou) [inline src/components/ProductReleaseCreateEditPage/ProductReleaseCreateEditPage.tsx:139]: also log this error to ui logger

File: src/components/ProductReleaseCreateEditPage/ProductReleaseCreateEditPage.tsx [hunk 4]
```diff
@@ -0,0 +1,368 @@
...
+      productPagesCode: data.productPagesCode,
+    });
+
+    return serviceContainerEditPagePatch
+      .run({ serviceData: { id: productReleaseId, patchData } })
+      .then(() => {
+        navigate('../..');
+      })
+      .catch((error: any) => {
+        console.error('Failed to edit Product Release.');
```
Comment (DnsZhou) [inline src/components/ProductReleaseCreateEditPage/ProductReleaseCreateEditPage.tsx:159]: also log this error to ui logger

**Review by DnsZhou — CHANGES_REQUESTED** [review]
Also rebase the code to the latest branch please.

File: src/components/ProductReleaseCreateEditPage/ProductReleaseCreateEditPage.tsx (hunk 3 above)
Comment (patrikk0123) [reply src/components/ProductReleaseCreateEditPage/ProductReleaseCreateEditPage.tsx:139]: I might, but we don't do that on other create/edit pages. @matedo1 What do you think?

Comment (matedo1) [reply src/components/ProductReleaseCreateEditPage/ProductReleaseCreateEditPage.tsx:140]: I would keep it as it is. I think it's useless to log it using uiLogger as there is nothing we can do to fix it - we just need to properly inform user that this action was not successful.

Comment (patrikk0123) [reply src/components/ProductReleaseCreateEditPage/ProductReleaseCreateEditPage.tsx:140]: And the user is informed by the `ServiceContainerCreatingUpdating`.

**Review by DnsZhou — APPROVED** [review]
changes lgtm, thx!

**Pull request merged**